src/auto-evo/simulation/MichePopulation.cs

```diff
@@ -44,12 +37,41 @@ public static class PopulationSimulation
 
         while (parameters.StepsLeft > 0)
         {
-            RunSimulationStep(parameters, speciesToSimulate, patchesList, random, cache,
-                parameters.AutoEvoConfiguration, parameters.WorldSettings);
+            RunSimulationStep(parameters, speciesToSimulate, patchesList, random, cache);
             --parameters.StepsLeft;
         }
     }
 
+    /// <summary>
+    ///   Estimates the initial population numbers
+    /// </summary>
+    public static int CalculateMicrobePopulationInPatch(MicrobeSpecies species, Miche miche,
+        BiomeConditions biomeConditions, SimulationCache cache)
+    {
+        var energyBalanceInfo = cache.GetEnergyBalanceForSpecies(species, biomeConditions);
+        var individualCost = energyBalanceInfo.TotalConsumptionStationary + energyBalanceInfo.TotalMovement
+            * species.Behaviour.Activity / Constants.MAX_SPECIES_ACTIVITY;
+
+        var leafNodes = miche.GetLeafNodes();
+
+        var miches = leafNodes.Where(x => x.Occupant == species).Select(x => x.BackTraversal())
```

Comment:
This LINQ chain seems like something that should be written out in code. Maybe the miche class could have this energy collection method?